### PR TITLE
wpcom-proxy-request: Return Promise if no callback specified (Take Two)

### PR DESCRIPTION
**Pull request by ockham**
Take Two of #39683. Reverts #39721.

- [x] TODO: Fix the issue documented in #39721.
 - The issue is only present in Docker. To repro, run `npm run distclean && npm ci && npm run build-docker`.

---

Original PR desc:

#### Changes proposed in this Pull Request

The promise wrapper introduced by @andrewserong et al. in #38823 seems generic enough for it to make sense to have it in `wpcom-proxy-request`. This PR moves it there. This follows [precedent](https://github.com/Automattic/wp-calypso/blob/bf345c5b040e75808fb2f92c9b9e34e6f126a235/packages/wpcom.js/src/lib/util/send-request.js#L78-L93) set by `wpcom.js`.

Note that this is a technically a breaking change, as noted in the package's `History.md`:

> - [...] Return Promise (rather than `XMLHttpRequest` instance) if no callback argument is provided.
>   - In practice, most people have probably been using the callback rather than the returned `XMLHttpRequest` instance, so this shouldn't be a breaking change for most.

#### Testing instructions

Verify that Gutenboarding still works (esp account and site creation).

### CODE REVIEW

**Review by sirreal — APPROVED** [review]
Let's try again!

File: packages/wpcom-proxy-request/package.json [hunk 1]
```diff
@@ -30,9 +30,11 @@
 	},
 	"files": [
 		"dist",
+		"types",
 		"History.md",
 		"README.md"
 	],
+	"types": "types",
```
[the remark this answers is not in the document]
Comment (ockham) [reply packages/wpcom-proxy-request/package.json]: Possibly relevant: 

> Also note that if your package.json includes the `"files"` property the `"types"` property will be ignored. In that case you need to pass your main declaration file to the `"files"` property as well.

(https://www.typescriptlang.org/docs/handbook/declaration-files/publishing.html)

[the remark this answers is not in the document]
Comment (ockham) [reply packages/wpcom-proxy-request/package.json]: So I probably should just remove the `"types"` property.

[the remark this answers is not in the document]
Comment (ockham) [reply packages/wpcom-proxy-request/package.json]: Hmm, fails at `npm ci` without :slightly_frowning_face:

[the remark this answers is not in the document]
Comment (sirreal) [reply packages/wpcom-proxy-request/package.json]: > Also note that if your package.json includes the `"files"` property the `"types"` property will be ignored. In that case you need to pass your main declaration file to the `"files"` property as well.

I wonder if that's accurate… seems pretty odd.

> Hmm, fails at `npm ci` without 🙁

Right, if we don't specify where the typings are, the other module can't find them and will fail with the implicit any on the module type.

**Pull request merged**